CRM/MembershipExtras/Hook/Post/MembershipPayment.php

```diff
@@ -118,4 +141,45 @@ private function getRelatedRecurringLineItem() {
     return [];
   }
 
+  /**
+   * Creates membership Period and associates it to appropriate payment entity.
+   */
+  private function updateMembershipPeriod() {
+    if ($this->periodExistsForMembershipPayment()) {
+      return;
+    }
+
+    $isSinglePayment = empty($this->contribution['contribution_recur_id']);
+
+    $paymentTable = $isSinglePayment
+      ? 'civicrm_contribution'
+      : 'civicrm_contribution_recur';
+
+    $entityID = $isSinglePayment
+      ? $this->contribution['id']
+      : $this->contribution['contribution_recur_id'];
+
+    CRM_MembershipExtras_BAO_MembershipPeriod::create([
+      'membership_id' => $this->membershipPayment->membership_id,
+      'start_date' => $this->membership['start_date'],
+      'end_date' => $this->membership['end_date'],
+      'payment_entity_table' => $paymentTable,
+      'entity_id' => $entityID,
+    ]);
+  }
+
+  /**
+   * Checks if the period already exists for the payment entity.
+   *
+   * @return bool
+   */
+  private function periodExistsForMembershipPayment() {
+    $paymentEntityID = $this->contribution['contribution_recur_id'] ?: $this->contribution['id'];
+

```

Comment:
no need for the space and I think it would be better to also check if the entity_type is correct